### PR TITLE
Various MSVC fixes

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

**Comment by tribal-tec** [comment]
retest this please

File: tests/file.cpp [hunk 1]
```diff
@@ -23,16 +23,17 @@
 
 int main( int, char** argv )
 {
-    const std::string& execPath = lunchbox::getExecutablePath();
     const boost::filesystem::path path( argv[0] );
-    const std::string argvPath( path.parent_path().string( ));
-    TEST( boost::algorithm::ends_with( execPath, argvPath ));
+    const std::string argvPath( path.parent_path().generic_string( ));
+    const boost::filesystem::path execPath( lunchbox::getExecutablePath( ));
+    std::cout << execPath.generic_string() << " " <<  argvPath << std::endl;
```
Comment (eile) [inline tests/file.cpp]: rm

**Comment by eile** [comment]
+1

**Comment by tribal-tec** [comment]
retest this please

**Pull request merged**